File: app/src/main/java/com/newage/plantedaqua/models/TankItems.java
package com.newage.plantedaqua.models;

public class TankItems {

   private String itemUri;
   private String txt1;
   private String txt2;
   private String txt3;
   private String txt4;
   private String tag;
   private Boolean isChecked = false;
   private String quickNote;
   private Boolean isShown = false;

    public Boolean getChecked() {
        return isChecked;
    }

    public void setChecked(Boolean checked) {
        isChecked = checked;
    }

    public Boolean getShown() {
        return isShown;
    }

    public void setShown(Boolean shown) {
        isShown = shown;
    }

    public String getQuickNote() {
        return quickNote;
    }

    public void setQuickNote(String quickNote) {
        this.quickNote = quickNote;
    }

    public String getTag() {
        return tag;
    }

    public void setTag(String tag) {
        this.tag = tag;
    }

    public String getItemUri() {
        return itemUri;
    }

    public void setItemUri(String itemUri) {
        this.itemUri = itemUri;
    }

    public String getTxt1() {
        return txt1;
    }

    public void setTxt1(String txt1) {
        this.txt1 = txt1;
    }

    public String getTxt2() {
        return txt2;
    }

    public void setTxt2(String txt2) {
        this.txt2 = txt2;
    }

    public String getTxt3() {
        return txt3;
    }

    public void setTxt3(String txt3) {
        this.txt3 = txt3;
    }

    public String getTxt4() {
        return txt4;
    }

    public void setTxt4(String txt4) {
        this.txt4 = txt4;
    }
}

File: app/src/main/java/com/newage/plantedaqua/models/TankItems.kt
package com.newage.plantedaqua.models

import java.util.*

//TankItems(I_ID text,I_Name text,I_Category text,I_URI text,I_Currency text,I_Price real,I_Quantity integer,I_BuyDate text,I_ExpDate text,I_Gender text,I_Food text,I_Care text,I_Quality text,I_Remarks text,I_Sci_Name text,I_Seller text)");
data class TankItems(

        var itemUri: String = "",
        var txt1: String = "",
        var txt2: String = "",
        var txt3: String = "",
        var txt4: String = "",
        var tag: String = "",
        var checked: Boolean = false,
        var quickNote: String = "",
        var shown: Boolean = false,
        var itemName: String = "",
        var itemCategory: String = "",
        var Currency: String = "",
        var itemPrice: String = "0.0",
        var itemQuantity: String = "1",
        var itemPurchaseDate: String = "",
        var itemExpiryDate: String = "",
        var itemGender : String = "",
        var food: String = "",
        var itemCare: String = "",
        var itemQuality: String = "1",
        var itemRemarks : String = "",
        var itemSeller : String ="",
        var itemScientificName : String = "",
        var mode : String = "creation"


){

    fun getNumericPrice() = (if(itemPrice.isBlank()) 0.0f else itemPrice.replace(",",".").toFloat())
    fun getNumericQuantity() = (if(itemQuantity.isBlank()) 1 else itemQuantity.toInt())
    fun String.returnOneIfBlank() = if(this.isBlank()) "1" else this
    fun String.returnZeroIfBlank() = if(this.isBlank()) "0.0" else this

    fun getTotalPrice(quantity : Int, price : Float) = String.format(Locale.getDefault(),"%.2f",(quantity*price))
}
